zenodo/modules/records/serializers/schemas/common.py

```diff
@@ -459,6 +460,50 @@ def postload_references(self, data):
                 for ref in data['references'] if ref.strip()
             ]
 
+    custom = fields.Method('dump_custom', 'load_custom')
+
+    def load_custom(self, obj):
+        if not obj:
+            return missing
+        else:
+            custom_metadata_rules = \
+                current_app.config.get('ZENODO_CUSTOM_METADATA_DEFINITIONS')
+            for community in obj.keys():
```

Comment:
No need for `.keys()` here (also below), `dict`s iterate over keys by default. 

`.items()` could be used though to avoid doing `obj[community]` below.